File: Chatgpt.java
import javax.swing.*;
import java.awt.*;
import ai.openai.*;
import java.util.*;

public class OpenAITechTeams {
    public static void main(String[] args) {
        OpenAI.apiKey = "Enter-your-api-key";

        String startSequence = "\nAI:";
        String restartSequence = "\nHuman: ";

        JFrame frame = new JFrame("OpenAI TechTeams");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        // User input
        JLabel userInputLabel = new JLabel("Enter your question:");
        userInputLabel.setFont(new Font("Poppins", Font.PLAIN, 16));
        userInputLabel.setPreferredSize(new Dimension(500, 30));
        JTextField userInput = new JTextField(60);
        userInput.setFont(new Font("NUnit Sans", Font.PLAIN, 12));
        userInput.setBackground(Color.YELLOW);

        // Generate button
        JButton generateButton = new JButton("Generate response");

        // Response output
        JLabel responseOutputLabel = new JLabel("AI's response:");
        responseOutputLabel.setFont(new Font("Poppins", Font.PLAIN, 16));
        responseOutputLabel.setPreferredSize(new Dimension(1000, 30));
        JTextArea responseOutput = new JTextArea();
        responseOutput.setEditable(false);
        responseOutput.setLineWrap(true);
        responseOutput.setWrapStyleWord(true);
        responseOutput.setBackground(Color.GRAY);
        responseOutput.setForeground(Color.WHITE);
        responseOutput.setFont(new Font("Poppins", Font.BOLD, 10));

        // Add components to frame
        JPanel panel = new JPanel(new FlowLayout());
        panel.add(userInputLabel);
        panel.add(userInput);
        panel.add(generateButton);
        panel.add(responseOutputLabel);
        JScrollPane scrollPane = new JScrollPane(responseOutput);
        scrollPane.setPreferredSize(new Dimension(1000, 500));
        panel.add(scrollPane);
        frame.getContentPane().add(panel);

        // Generate response when button is clicked
        generateButton.addActionListener(e -> {
            String ask = userInput.getText();
            if (ask.equals("break")) {
                responseOutput.append("thank you: ");
            } else {
                Completion response = Completion.create(
                    Model.Davinci,
                    ask,
                    new CompletionParameters.Builder()
                        .setTemperature(0.9)
                        .setMaxTokens(150)
                        .setTopP(1)
                        .setFrequencyPenalty(0)
                        .setPresencePenalty(0.6)
                        .addStop(" Human:")
                        .addStop(" AI:")
                        .build()
                );
                String responseText = response.getChoices().get(0).getText();
                responseOutput.append(startSequence + responseText + restartSequence);
            }
            responseOutput.setCaretPosition(responseOutput.getDocument().getLength());
        });

        frame.pack();
        frame.setVisible(true);
    }
}

File: Chatgpt.py
import tkinter as tk
import openai

openai.api_key = "Your_API_KEY"

start_sequence = "\nAI:"
restart_sequence = "\nHuman: "


def generate_response():
    ask = user_input.get()
    if ask == "break":
        response_output.config(state=tk.NORMAL)
        response_output.insert(tk.END, "thank you: ")
        response_output.config(state=tk.DISABLED)
    else:
        response = openai.Completion.create(
            model="text-davinci-003",
            prompt=ask,
            temperature=0.9,
            max_tokens=150,
            top_p=1,
            frequency_penalty=0,
            presence_penalty=0.6,
            stop=[" Human:", " AI:"]
        )
        response_output.config(state=tk.NORMAL)
        response_output.insert(tk.END, f"{start_sequence}{response['choices'][0]['text']}{restart_sequence}")
        response_output.config(state=tk.DISABLED)


root = tk.Tk()
root.title("OpenAI Chatbot")

# User input
user_input_label = tk.Label(root, text="Enter your question:")
user_input_label.pack()
user_input = tk.Entry(root)
user_input.pack()

# Generate button
generate_button = tk.Button(root, text="Generate response", command=generate_response)
generate_button.pack()

# Response output
response_output_label = tk.Label(root, text="AI's response:")
response_output_label.pack()
response_output = tk.Text(root, state=tk.DISABLED)
response_output.pack()

root.mainloop()
